### PR TITLE
[Improve][Connector-V2][Redis] RedisSource support scan operation

### PR DESCRIPTION
**Pull request by rhh777**
Scan command allow for incremental iteration, returning only a small number of elements per call,.
KEYS command may block the server for a long time when called against a lot of keys.
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

**Comment by TyrantLucifer** [comment]
It looks like a exciting feature. But it seems that has some problems in e2e test case. BTW, if you write a e2e test case to verify this feature is better. A tested PR will join the host branch more quickly. Thank you again for your contribution to SeaTunnel.

**Comment by rhh777** [comment]
> It looks like a exciting feature. But it seems that has some problems in e2e test case. BTW, if you write a e2e test case to verify this feature is better. A tested PR will join the host branch more quickly. Thank you again for your contribution to SeaTunnel.

I unit tested it locally, but I'm still learning e2e, so I'll fill in later.

File: seatunnel-connectors-v2/connector-redis/src/main/java/org/apache/seatunnel/connectors/seatunnel/redis/config/RedisConfig.java [hunk 1]
```diff
@@ -60,7 +61,19 @@ public enum HashKeyParseMode {
         Options.key("keys")
             .stringType()
             .noDefaultValue()
-            .withDescription("keys pattern, redis source connector support fuzzy key matching, user needs to ensure that the matched keys are the same type");
+            .withDescription("keys pattern, redis source connector support fuzzy key matching, if keys_type_check is false, user needs to ensure that the matched keys are the same type");
+
+    public static final Option<Boolean> KEY_TYPE_CHECK =
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-redis/src/main/java/org/apache/seatunnel/connectors/seatunnel/redis/config/RedisConfig.java:65]: Please update the document and add change log about this pr.

Comment (EricJoy2048) [reply seatunnel-connectors-v2/connector-redis/src/main/java/org/apache/seatunnel/connectors/seatunnel/redis/config/RedisConfig.java:65]: You need update the RedisSourceFactory too.

Comment (rhh777) [reply seatunnel-connectors-v2/connector-redis/src/main/java/org/apache/seatunnel/connectors/seatunnel/redis/config/RedisConfig.java:65]: @EricJoy2048 Update completed

File: docs/en/connector-v2/source/Redis.md [hunk 2]
```diff
@@ -253,6 +263,10 @@ simple:
 
 - Add Redis Source Connector
 
+### 2.3.1-SNAPSHOT 2023-01-05
```
Comment (EricJoy2048) [inline docs/en/connector-v2/source/Redis.md:270]: Add this change log to `next version` please. @TyrantLucifer  PTAL.

Comment (EricJoy2048) [reply docs/en/connector-v2/source/Redis.md:270]: And you need add the release note to `https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md` too.

**Comment by hailin0** [comment]
check code style

**Comment by EricJoy2048** [comment]
Hi, @rhh777  
 You submitted a very good pr, but I found that it has not been changed for a long time, can you update this PR?

**Comment by FuYouJ** [comment]
Please assign it to me

**Comment by EricJoy2048** [comment]
> Please assign it to me

Ok.

**Pull request closed without merging**